newsroom/core/resources/validators.py

```diff
@@ -62,3 +64,29 @@ def _validate_dashboard_type_exists(value: str | None = None) -> str | None:
         return value
 
     return AfterValidator(_validate_dashboard_type_exists)
+
+
+def validate_multi_field_iunique_value_async(resource_name: str, field_names: list[str]) -> AsyncValidator:
+    """Validate that the combination of fields is unique in the resource (case-insensitive)
+
+    :param resource_name: The name of the resource where the combination of fields must be unique
+    :param field_names: The names of the fields to ensure unique combination
+    """
+
+    async def _validate_iunique_value(item: ResourceModel, value: Any) -> None:
+        fields = {field: getattr(item, field) for field in field_names}
+        if any(value for value in fields.values()):
```

Comment:
What is the purpose of this if/return statement?

Reply:
my bad, I changed this last moment, it should have been 
```
if any(value is None for value in fields.values()):
```
I'll fix it now.